core/store/migrate/migrations/0081_unconsumed_log_broadcasts.sql

```diff
@@ -0,0 +1,17 @@
+-- +goose Up
+ALTER TABLE log_broadcasts ADD COLUMN updated_at timestamp with time zone NOT NULL;
+DROP INDEX IF EXISTS log_consumptions_unique_v2_idx;
+CREATE UNIQUE INDEX log_broadcasts_unique_idx ON log_broadcasts(job_id, block_hash, log_index, evm_chain_id);
+CREATE TABLE log_broadcasts_pending (
+    evm_chain_id numeric(78,0) REFERENCES evm_chains (id) DEFERRABLE INITIALLY IMMEDIATE PRIMARY KEY,
+    block_number int8,
+    created_at timestamp with time zone NOT NULL,
+    updated_at timestamp with time zone NOT NULL
+);
```

Comment:
I originally tried to be clever with `job_id = null` rows in the existing table, but all we really need is the minimum block number and we can track that directly with fewer writes anyways.

Reply:
Makes sense. One alternate design maybe worth some consideration: could we simplify by changing the log_broadcasts table into more of a simple "log" table with a `status` field which is one of {received, broadcasted, consumed}? We'd write the log immediately to the pool and the db upon receiving it as status=received. On node restart we'd rebuild the pool from the db based on the log records which are not marked consumed. When there's a reorg and we get a log with Removed=true, we delete it from the db. Mark it as status=broadcasted when we broadcast and then status=consumed when its consumed.

Reply:
It might be possible to do something like this, but I think it would have to include a larger refactor of the broadcaster internals (e.g. creating a row per subscriber job up front?). Something else I tried to account for that could be a problem is that we can't depend on receiving removed logs, so we'd have to verify any data we pulled back out of the DB anyways. It seemed cleaner to just extend the backfill to replay everything and ignore what we already consumed, rather than verify all of the individual leftover logs.

Reply:
yeah agreed - larger refactor